sycl/doc/extensions/DataFlowPipes/data_flow_pipes.asciidoc

```diff
@@ -0,0 +1,650 @@
+  template <unsigned ID>
+  struct ethernet_pipe_id {
+    static constexpr unsigned id = ID;
+  };
+  
+  using ethernet_read_pipe = kernel_readable_io_pipe<ethernet_pipe_id<0>, int, 0>;
+  using ethernet_write_pipe =kernel_writeable_io_pipe<ethernet_pipe_id<1>, int, 0>;
+}
+----
+
+== Memory model
+
+Pipes in the context of this extension step outside the OpenCL and SYCL memory models in the following ways:
+
+. Pipes are not defined to be in any address space.  Each pipe is conceptually in its own address space, that does not alias with any others.
+
+. Data written to a pipe must be eventually visible to the read endpoint of the pipe without an OpenCL synchronization point.  Specifically, kernel completion or other synchronization points are not required to guarantee pipe write side effect visibility on the read endpoint of the pipe.
+
+. There is no implicit synchronizes-with relationship between different pipes and/or with non-pipe memory in a named address space (e.g. global, local, private).  Specifically, there is no implicit global or local release of side effects through a pipe access, and observation of data or control information on one pipe does not imply any knowledge through happens-before relationship with a different pipe or with memory not associated with the pipe.
+
+. At a work-group barrier, there is an implicit acquire and release of side effects for any pipes operated on within the kernel, either before or after the barrier.  This occurs without an explicit pipe memory fence being applied as a barrier operand.
+
+. There are no guarantees on pipe operation side effect latency.  Writes to a pipe will eventually be visible to read operations on the pipe, without a synchronization point, but that visibility is not guaranteed to be by the time that the next instruction is executed by a writing work-item, for example.  There may be arbitrary latency between a write to a pipe and visibility of the data on a read endpoint of the pipe.  Likewise, there may be arbitrary latency between a read from a pipe, and visibility at a write endpoint that there is capacity available to write to (assuming that capacity was full prior to the read).
+
+. Data in a pipe exists within an instance of a program object on a device (programming bitstream lifetime on FPGA devices).  Invocation of a kernel from a different program object destroys all data stored in pipes within the program object associated with the previous kernel(s) executed on the device.  Different devices using the same SYCL program object maintain independent pipe state and data.
+
+
+== SYCL DAG and pipes
+The SYCL `accessor` mechanism exposes an object through which to access data, but also adds dependency edges (_requirements_) between nodes in the DAG.  Pipes imply no such dependency edges, regardless of whether they connect purely kernels, or also to host pipe operations.  A user should create DAG ordering constraints, using events for example, if required.
```

Comment:
Yes, pipes are one of the options for implicit DAG edges, but that's for a different extension for a few reasons :)

I don't think the different read/write size is possible in this extension interface (at the SYCL level), but yes, it's possible at the SPIR-V level with unclear semantics.  SYCL code can manually do width adaptation, but we're not trying to capture the mixed read/write width case through the pipe interface, since I believe that it normally signals a programming error.  Interesting observation worth considering capturing at the SYCL level if enough use cases arise.